src/base/dense/owl_base_dense_ndarray_generic.ml

```diff
@@ -760,7 +760,47 @@ let max' varr = (_fold_left (Pervasives.max) Pervasives.min_float varr)
 
 
 (* Sum of all elements *)
-let sum' varr = (_fold_left (+.) 0. varr)
+let sum' varr =
+  let _kind = kind varr in
+  _fold_left (Owl_base_dense_common._add_elt _kind) (Owl_const.zero _kind) varr
+
+
+(** Folding along a specified axis, aka reduction.
+    m: number of slices; n: x's slice size; o: x' strides, also y's slice size;
+    x: source; y: shape of destination. Note that o <= n. *)
+let owl_sum_along add_fun m n o x ys =
```

Comment:
OK, I also meant that this owl_sum_along can be generalised as fold_along, which allows you to plug in not only sum, but aslo min, max, prod, and etc., so that you can further implement min_reduce, max_reduce in future if necessary.

Reply:
Oh I see. I think we can pass it any function of type `a'->a'->a'` to it to conduct fold operations.  I renamed this function and add TODO notes.